src/usr/local/www/widgets/widgets/ipsec.widget.php

```diff
@@ -125,58 +125,48 @@
 	}
 
 	// Generate JSON formatted data for the widget to update from
-	$jsondata = "{";
-
-	$jsondata .= "\"overview\":\"";
-	$jsondata .= "<tr>";
-	$jsondata .= "<td>" . $activecounter . "</td>";
-	$jsondata .= "<td>" . $inactivecounter . "</td>";
-	$jsondata .= "<td>" . (is_array($mobile['pool']) ? htmlspecialchars($mobile['pool'][0]['usage']) : '0') . "</td>";
-	$jsondata .= "</tr>";
-	$jsondata .= "\",\n";
-
-	$jsondata .= "\"tunnel\":\"";
+	$data = new stdClass();
+	$data->overview = "<tr>";
+	$data->overview .= "<td>" . $activecounter . "</td>";
+	$data->overview .= "<td>" . $inactivecounter . "</td>";
+	$data->overview .= "<td>" . (is_array($mobile['pool']) ? htmlspecialchars($mobile['pool'][0]['usage']) : '0') . "</td>";
+	$data->overview .= "</tr>";
+
+	$data->tunnel = "";
 	if(is_array($ipsec_detail_array) && !empty($ipsec_detail_array)){
 		foreach ($ipsec_detail_array as $ipsec) {
-			$jsondata .= "<tr>";
-			$jsondata .= "<td>" . htmlspecialchars($ipsec['src']) . "</td>";
-			$jsondata .= "<td>" . $ipsec['remote-subnet'] . "<br />(" . htmlspecialchars($ipsec['dest']) . ")</td>";
-			$jsondata .= "<td>" . htmlspecialchars($ipsec['descr']) . "</td>";
-
+			$data->tunnel .= "<tr>";
+			$data->tunnel .= "<td>" . htmlspecialchars($ipsec['src']) . "</td>";
+			$data->tunnel .= "<td>" . $ipsec['remote-subnet'] . "<br />(" . htmlspecialchars($ipsec['dest']) . ")</td>";
+			$data->tunnel .= "<td>" . htmlspecialchars($ipsec['descr']) . "</td>";
 			if ($ipsec['status'] == "true") {
-				$jsondata .= '<td><i class=\"fa fa-arrow-up text-success\"></i></td>';
+				$data->tunnel .= '<td><i class=\"fa fa-arrow-up text-success\"></i></td>';
 			} else {
-				$jsondata .= '<td><i class=\"fa fa-arrow-down text-danger\"></i></td>';
+				$data->tunnel .= '<td><i class=\"fa fa-arrow-down text-danger\"></i></td>';
```

Comment:
These double quotes no longer need escaped, they are getting escaped again which breaks the status icon.